beacon-chain/blockchain/chain_info.go

```diff
@@ -95,6 +103,12 @@ func (s *Service) HeadState() *pb.BeaconState {
 	return proto.Clone(s.headState).(*pb.BeaconState)
 }
 
+// BlockExists checks if a block with the provided root exists
+// in the db.
+func (s *Service) BlockExists(ctx context.Context, root [32]byte) bool {
```

Comment:
Why do you need to go through chain service for checking if a block exists in db? it seems like an extra layer. In this example `InitialSync` can have access to DB?